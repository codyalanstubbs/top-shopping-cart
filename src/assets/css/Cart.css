.cart-count {
    position: fixed;
    margin: -10px 0 0 -20px;
}