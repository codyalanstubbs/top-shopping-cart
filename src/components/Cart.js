import React, { useState } from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCartShopping } from '@fortawesome/free-solid-svg-icons';
import "../assets/css/Cart.css";

const Cart = ({shoppingCart}) => {
    // Calculate total quantity of all product quantities
    let numItemsInCart = shoppingCart.reduce((totalQty, product) => {
            return totalQty + product.qty;
    }, 0);

    return (
        <div className="cart">
            <FontAwesomeIcon icon={faCartShopping}/>
            <div data-testid="cart-count" className="cart-count">
                {numItemsInCart}
            </div>
        </div>
    );
};

export default Cart;