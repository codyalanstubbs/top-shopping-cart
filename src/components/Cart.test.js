import React from 'react';
import uniquid from "uniquid";
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Cart from './Cart';

const defaultShoppingCart = [];

describe('Cart', () => {
    
    describe('Render default Cart', () => {
        
        test("Renders Cart snapshot", () => {
            const utils = render(<Cart shoppingCart={defaultShoppingCart}/>);
            expect(utils).toMatchSnapshot();
        });   

        test("Renders Cart with cart-count 0", () => {
            render(<Cart shoppingCart={defaultShoppingCart}/>);

            const cartCount = screen.getByTestId("cart-count");
            expect(cartCount).toBeInTheDocument();
            expect(cartCount.textContent).toBe("0");
        });

    });

    describe('Render Cart with products', () => {
        
        test("Renders Cart with cart-count 0 despite having products in shoppingCart", () => {
            render(
                <Cart 
                    shoppingCart={
                        [
                            {id: uniquid(), name: "Apples",     cost: 100, qty: 0},
                            {id: uniquid(), name: "Oranges",    cost: 100, qty: 0},
                            {id: uniquid(), name: "Chocolate",  cost: 100, qty: 0},
                            {id: uniquid(), name: "Green Tea",  cost: 100, qty: 0},
                        ]
                    }
                />
            );

            const cartCount = screen.getByTestId("cart-count");
            expect(cartCount).toBeInTheDocument();
            expect(cartCount.textContent).toBe("0");
        });

        test("Renders Cart with cart-count 1 despite having multiple products in shoppingCart", () => {
            render(
                <Cart 
                    shoppingCart={
                        [
                            {id: uniquid(), name: "Apples",     cost: 100, qty: 0},
                            {id: uniquid(), name: "Oranges",    cost: 100, qty: 1},
                            {id: uniquid(), name: "Chocolate",  cost: 100, qty: 0},
                            {id: uniquid(), name: "Green Tea",  cost: 100, qty: 0},
                        ]
                    }
                />
            );

            const cartCount = screen.getByTestId("cart-count");
            expect(cartCount).toBeInTheDocument();
            expect(cartCount.textContent).toBe("1");
        });

        test("Renders Cart with cart-count 2 despite having multiple products in shoppingCart", () => {
            render(
                <Cart 
                    shoppingCart={
                        [
                            {id: uniquid(), name: "Apples",     cost: 100, qty: 1},
                            {id: uniquid(), name: "Oranges",    cost: 100, qty: 1},
                            {id: uniquid(), name: "Chocolate",  cost: 100, qty: 0},
                            {id: uniquid(), name: "Green Tea",  cost: 100, qty: 0},
                        ]
                    }
                />
            );

            const cartCount = screen.getByTestId("cart-count");
            expect(cartCount).toBeInTheDocument();
            expect(cartCount.textContent).toBe("2");
        });

        test("Renders Cart with cart-count 2 with only 1 product in cart", () => {
            render(
                <Cart 
                    shoppingCart={
                        [
                            {id: uniquid(), name: "Apples",     cost: 100, qty: 2},
                        ]
                    }
                />
            );

            const cartCount = screen.getByTestId("cart-count");
            expect(cartCount).toBeInTheDocument();
            expect(cartCount.textContent).toBe("2");
        });

    });
    
});
