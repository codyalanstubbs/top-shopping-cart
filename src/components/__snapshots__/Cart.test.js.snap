// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Render default Cart Renders Cart snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cart"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cart-shopping "
          data-icon="cart-shopping"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
            fill="currentColor"
          />
        </svg>
        <div
          class="cart-count"
          data-testid="cart-count"
        >
          0
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cart"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-cart-shopping "
        data-icon="cart-shopping"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
          fill="currentColor"
        />
      </svg>
      <div
        class="cart-count"
        data-testid="cart-count"
      >
        0
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
